src/server.js

```diff
@@ -202,6 +210,48 @@ app.get(
   }
 );
 
+app.get(
+  `${PREFIX_API}/:${URL_PARAM_EVENT_ID}/restaurants`,
+  getEvent,
+  async (request, response) => {
+    const { event } = request;
+    const users = event.users || {};
+    const userInput = Object.values(users);
+
+    if (userInput.length > 0) {
+      const lat = userInput[0].lat.toString();
+      const long = userInput[0].long.toString();
+      let radius = "50000";
+      let type = "restaurant";
+      let minprice = "0";
+      let maxprice = "4";
+      const restarantData = await (
+        await fetch(
+          `https://maps.googleapis.com/maps/api/place/nearbysearch/json?location=${lat},${long}&radius=${radius}&type=${type}&minprice=${minprice}&maxprice=${maxprice}&key=${env.API_KEY}`
+        )
+      ).json();
+      try {
+        response.json({
+          status: 200,
+          data: restarantData,
+        });
+      } catch (err) {
+        console.error("Bad Places API interaction");
+        response.status(500).json({
+          status: 500,
+          error: { type: ERROR_BAD_PLACES_API_INTERACTION },
+        });
+      }
+    } else {
+      console.error("Failed to get restaurants. Undefined Location.");
```

Comment:
Why does this code path indicate an undefined location? Can we check for an undefined location earlier in the code and submit an error before hitting the API?

Reply:
**Edit**: Nevermind, I see that this is all part of a giant else block. One thing to consider is, there is a valid state where there is no user location data: the period of time before anyone has submitted anything. In that case, we should return an empty list of restaurants (since we don't know the location of anyone) instead of returning an error, since it's a valid part of the app flow.

Reply:
So, there was this problem. The first time I submit data to data-store (using the UI search button) it always returns an empty object. Do you have an idea why that is? I know you handled that in the getEvent function. I bascially put this if else block to handle that case

Reply:
Oh I see, you mention in a comment below that there is a valid state where no-one has submitted anything.  Even when I submit something, the first time is always an empty data-store object.